cluster-api/api/cluster/v1alpha1/types.go

```diff
@@ -38,10 +38,32 @@ type Cluster struct {
 	Status ClusterStatus `json:"status,omitempty"`
 }
 
+type CloudProviderName string
+
+const (
```

Comment:
I don't really have any strong opinions about the name/implementation here. Just would like something to switch on. Could be enumerated for all I care.